pyqtorch/matrices.py

```diff
@@ -3,7 +3,8 @@
 import torch
 from torch import Tensor
 
-DEFAULT_MATRIX_DTYPE = torch.cdouble
+DEFAULT_REAL_DTYPE = torch.get_default_dtype()
+DEFAULT_MATRIX_DTYPE = torch.cdouble if DEFAULT_REAL_DTYPE == torch.float64 else torch.cfloat
```

Comment:
This always makes me afarid of order of execution issues.

Is it not better to query for this when needed? Rather than here which will probably run on `import pyqtorch`?

Reply:
jep, good point

Reply:
not sure how to realize this exactly. i'd say that if the user wants to use higher precision, they should always put the `torch.set_default_dtype` call at the very start of their script.

Reply:
redone now so you pass the dtype like a device